python/triton/compiler/compiler.py

```diff
@@ -272,6 +272,8 @@ def compile(src, target=None, options=None):
         ir.load_dialects(context)
         backend.load_dialects(context)
 
+    if os.environ.get("MLIR_DISABLE_MULTITHREADING", "0") == "1":
+        context.disable_multithreading()
```

Comment:
I'm not sure how would this affect the ASAN path that also disables multithreading below. I guess if MT is disabled in the first place, asan might just be fine? @CRobeck is there a way that I can test this together with ASAN?

Reply:
The test seems failing already:

`python3 -m pytest third_party/amd/python/test/test_address_sanitizer.py`

>       assert "Begin function __asan_report" in out.stdout.read().decode()
E       AssertionError: assert 'Begin function __asan_report' in ''

Reply:
Are you running on the NV side? Is only supported on AMD (ASAN isn't supported on the NV backend). We did turn this test off due to a regression maybe a week ago and haven't had a chance to look at it. cc: @antiagainst

Reply:
It is actually on MI300.

Reply:
OK, then it is probably missing the ASAN libraries. See here:
https://github.com/triton-lang/triton/blob/main/.github/workflows/integration-tests.yml.in#L410
it's kind of pain to get going, you need to have the ASAN libraries on the system you're running it on.